### ISSUE TITLE
M3U playlist shows up empty.

### ISSUE THREAD

**Issue by jimmiedave** [ejurgensen/forked-daapd#939]
Using v. 27.1, Debian Buster, built locally on an ARM based QNAP NAS, serving 18,000 songs over LAN, basically working fine in most other respects. 

But I can't get a list of radio presets, contained in a M3U file, to appear in the Playlists or daap shared library.

I have:

- typed playlists
- exported playlists from Music.app (the successor to iTunes)
- re-saved in text editor and/or used mac2unix to make sure I had linux-flavored end-of-line characters
- made sure there were no https stream links
- tried with and without #EXTM3U tags
- copied other people's playlists from the git issues
- enabled and disabled "radio" in forked-daapd.conf

...and I can't get a playlist to show up with its radio stream entries in Music.app.

I send the playlist up to the server with rsync, then forked-daapd sees it and processes it, but I end up with only the playlist name when I reload the daap library. 

When I click on the name, Music.app says: "Empty Playlist. To add to this playlist, drag items to it..."

![Screen Shot 2020-04-11 at 9 39 10 PM](https://user-images.githubusercontent.com/35353530/79058526-360ea480-7c3d-11ea-98d5-0c581e4182ad.png)

It gives the playlist the name of the file (even though I've used the "#PLAYLIST" tag to specify the playlist name), and says "NO ITEMS".

Here's the playlist entry in songs3.db:

```
sqlite> select * from playlists where title like 'Radio%';

25|Radio_Faves|3|(NULL)|1586654086|0|/home/forked-daapd/mp3s/Radio_Faves/Radio_Faves.m3u|0|0|/file:/home/forked-daapd/mp3s/Radio_Faves/Radio_Faves|0|12335||0|1

```
Here's the latest playlist I've tried:

```
#EXTM3U
#EXTINF:-1,WABE Live Stream - 
http://pba-ice.streamguys.org:80/wabe
#EXTINF:-1,WAMU Live Stream - 
http://wamu-1.streamguys.com:80
#EXTINF:-1,KCRW Eclectic Music - 
http://kcrw.streamguys1.com/kcrw_192k_mp3_e24_internet_radio
#EXTINF:-1,KCRW Live Stream - 
http://kcrw.streamguys1.com/kcrw_192k_mp3_on_air_internet_radio
#EXTINF:-1,Radio Paradise (aac-128) -  
http://stream-uk1.radioparadise.com:80/aac-128
http://bbcmedia.ic.llnwd.net/stream/bbcmedia_radio1_mf_p
http://bbcmedia.ic.llnwd.net/stream/bbcmedia_radio2_mf_p
http://bbcmedia.ic.llnwd.net/stream/bbcmedia_radio4fm_mf_p
http://bbcmedia.ic.llnwd.net/stream/bbcmedia_radio4extra_mf_p
```

Any idea why I can't get this working?

**Comment by Porco-Rosso** [comment]
Have you tried a simple .m3u with just one radio stream? 
My m3u file containing just this one line works as expected and shows up as a playlist:
`https://listen.radioking.com/radio/213478/stream/256591` 

On a slight tangent. My other m3u playlists were also coming up empty on forked-daapd after one of the updates, but in my case all I had to do was rename them from .m3u8 to .m3u. Could we get support for m3u8 files?

**Comment by Porco-Rosso** [comment]
Actually, upon checking the playlists through Music, and not the web-ui I now also realize my radio streams are all missing completely. Prob need to doublecheck the config section dealing with radio channels...

**Comment by ejurgensen** [comment]
See issue #410 for an explanation of this

**Comment by ejurgensen** [comment]
As you can see from the above issue the current design does not support radio streams in iTunes. It is probably possible to add this, but it is not something I have prioritized so far. I will close this issue since it is duplicate of 410. You are welcome to ping me on that issue in some months if it is still not done.

**Comment by jimmiedave** [comment]
I understand reason for the close, though my expectation was not that forked-daapd would stream radio.

I assumed that since the doc said it supports M3U playlists, that the contents of the playlist would appear. It'd be up to the client device or software to follow the streaming link to get the stream.

My use case: I don't keep 18,000 songs filling up my laptop's hard disk. They live on my music server. I don't buy anything from the iTunes music store. Thus, the only thing I do with iTunes/Music.app is stream from forked-daapd for my music library, and listen to network radio.

It's a pain to have a few clicks to switch back and forth between the local iTunes/Music.app library (where my typical radio stations live in a single playlist) and the forked-daapd shared library, where all my music lives.

This has become extra-apparent during the Covid-19 social distancing, because I'm fumbling with switching libraries a number of times a day, and would just like "all my stuff" in one place.

I definitely don't need radio streaming/transcoding from forked-daapd, just a playlist.

You could take the issue as a doc fix at a minimum.

**Comment by ejurgensen** [comment]
> I assumed that since the doc said it supports M3U playlists, that the contents of the playlist would appear. It'd be up to the client device or software to follow the streaming link to get the stream.

It's not a problem to make the http items appear (they are currently removed on purpose), the non-trivial part is to make them work. With daap, the server does not just give a link to the client.

I'm sure it is possible to support, but it would take some work to find out how.

I agree the doc should say this, I have fixed this with commit c43e009.
